solvation_analysis/analysis.py

```diff
@@ -0,0 +1,321 @@
+                a solvation radius as output. e.g. rdf_kernel(bins, data) -> 3.2. By default,
+                the rdf_kernel is solvation_analysis.rdf_parser.identify_solvation_cutoff.
+            kernel_kwargs : dict
+                kwargs passed to rdf_kernel
+            rdf_init_kwargs : dict
+                kwargs passed to inner rdf initialization
+            rdf_run_kwargs : dict
+                kwargs passed to inner rdf run e.g. inner_rdf.run(**rdf_run_kwargs)
+            kwargs : dict
+                kwargs passed to AnalysisBase
+        """
+        super(Solution, self).__init__(solute.universe.trajectory, **kwargs)
+        self.radii = {} if radii is None else radii
+        self.kernel = identify_solvation_cutoff if rdf_kernel is None else rdf_kernel
+        self.kernel_kwargs = {} if kernel_kwargs is None else kernel_kwargs
+        self.rdf_init_kwargs = {"range": (0, 8.0)} if rdf_init_kwargs is None else rdf_init_kwargs
+        self.rdf_run_kwargs = {} if rdf_run_kwargs is None else rdf_run_kwargs
+        # TODO: save solute numbers somewhere
+        self.solute = get_atom_group(solute)
+        self.n_solute = len(self.solute)
+        self.solvents = solvents
+        self.u = self.solute.universe
+        self.rdf_plots = {}
+        self.rdf_data = {}
+        self.ion_speciation = None
+        self.ion_pairing = None
+        self.coordination_numbers = None
+        self.solvation_frames = []
+
+    @classmethod
+    def _plot_solvation_radius(cls, bins, data, radius):
+        """
+        Will plot the solvation radius on the rdf. If
+
+        Parameters
+        ----------
+            bins : np.array
+                the rdf bins
+            data : np.array
+                the rdf data
+            radius : float
+                the cutoff radius to draw on the plot
+
+        Returns
+        -------
+            Matplotlib Figure, Matplotlib Axis
+        """
+        fig, ax = plt.subplots()
+        ax.plot(bins, data, "b-", label="rdf")
+        ax.axvline(radius, color="r", label="solvation radius")
+        ax.set_xlabel("Radial Distance (A)")
+        ax.set_ylabel("Probability Density")
+        ax.legend()
+        return fig, ax
+
+    def _prepare(self):
+        """
+        This function identifies the solvation radii and saves the associated rdfs.
+        """
+        for name, solvent in self.solvents.items():
+            # generate and save RDFs
+            rdf = InterRDF(self.solute, solvent, **self.rdf_init_kwargs)
+            # TODO: specify start and stop args
+            rdf.run(**self.rdf_run_kwargs)
+            bins, data = rdf.results.bins, rdf.results.rdf
+            self.rdf_data[name] = (data, bins)
+            # generate and save plots
+            if name not in self.radii.keys():
+                self.radii[name] = self.kernel(bins, data, **self.kernel_kwargs)
+            fig, ax = self._plot_solvation_radius(bins, data, self.radii[name])
```

Comment:
The more I think about it I am not sure that plotting by default in `_prepare` is a good idea. This kind of goes against the idea that `_prepare` `_run` and `_conclude` should be as minimal as possible and deal primarily with data.

I think we should move the majority of the plotting logic to a separate method. What do you think @orioncohen? @richardjgowers? I could be convinced either way.

Reply:
Are you suggesting that we move the plotting logic to another method and then call that method in prepare? If so, that sounds good to me. 

Otherwise, I would be hesitant to force the user to manually call a plotting method before calling `_run`, that was the design I had earlier.

Reply:
I think the basic approach would be to have (1) data generation and (2) plotting of data. (1) should be completely independent of (2). Plotting obviously needs the data (but in an ideal world you could provide the data by any means.)

Reply:
I was thinking after! you don't need to see the plots until you "request" them with something like. `Solution.rdf_plots()` so there is no need to precompute them. The data is all stored internally so just make the plots on demand?

Reply:
Got it. I misunderstood @hmacdope's suggestion. I'll pull the plotting out to another method!